### PR TITLE
Implement `time` type using chrono and date

### PR DESCRIPTION
**Pull request by brainiac**
Added support for milliseconds, based on work started by @martinlindhe

- Added the following members to the time type: Milliseconds, MillisecondsSinceEpoch and MillisecondsSinceMidnight
- Set night time range to [7pm, 5am)

Times are represented as local time, otherwise we have to account for time zones.

### CODE REVIEW

**Comment by brainiac** [comment]
I def pushed the wrong changes, hold on let me fix that.

**Comment by brainiac** [comment]
alright fixed. note to self: don't amend a merge commit and then attempt to rebase it.

File: src/main/MQ2Utilities.cpp [hunk 1]
```diff
@@ -1250,7 +1250,7 @@ void GetGameTime(int* Hour, int* Minute, int* Night)
 	if (Minute)
 		*Minute = eqMinute;
 	if (Night)
-		*Night = ((eqHour < 7) || (eqHour > 18));
+		*Night = eqHour >= 19 || eqHour < 5; // [7pm, 5am)
```
Comment (dannuic) [inline src/main/MQ2Utilities.cpp:1253]: This varies per server at least, doesn't it? Maybe we should have a night/day cycle config in the future.

**Review by dannuic — APPROVED** [review]
LGTM

**Pull request merged**